src/object/cli_obj.c

```diff
@@ -1834,30 +1834,28 @@ dc_obj_layout_refresh(daos_handle_t oh)
 }
 
 uint32_t
-dc_obj_retry_delay(tse_task_t *task, int err, uint16_t *retry_cnt, uint16_t *inprogress_cnt,
-		   uint32_t timeout_sec)
-{
-	uint32_t	delay = 0;
-	uint32_t	limit = 4;
-
-	/*
-	 * Randomly delay 5 ~ 1028 us if it is not the first retry for
-	 * -DER_INPROGRESS || -DER_UPDATE_AGAIN cases.
-	 */
-	++(*retry_cnt);
-	if (err == -DER_INPROGRESS || err == -DER_UPDATE_AGAIN || err == -DER_NOTLEADER) {
-		if (++(*inprogress_cnt) > 1) {
-			limit += *inprogress_cnt;
-			if (limit > 10)
-				limit = 10;
-
-			delay = (d_rand() & ((1 << limit) - 1)) + 5;
-			/* Rebuild is being established on the server side, wait a bit longer */
-			if (err == -DER_UPDATE_AGAIN)
-				delay <<= 10;
-			D_DEBUG(DB_IO, "Try to re-sched task %p for %d/%d times with %u us delay\n",
-				task, (int)*inprogress_cnt, (int)*retry_cnt, delay);
-		}
+dc_obj_retry_delay(tse_task_t *task, uint32_t opc, int err, uint16_t *retry_cnt,
+		   uint16_t *inprogress_cnt, uint32_t timeout_sec)
+{
+	uint32_t delay = 0;
+
+	if (err == -DER_INPROGRESS || err == -DER_UPDATE_AGAIN || err == -DER_NOTLEADER)
+		++(*inprogress_cnt);
+
+	if (++(*retry_cnt) > 1) {
+		/* Randomly delay [31 ~ 1023] us if it is not the first retried object RPC. */
+		delay = (d_rand() | ((1 << 5) - 1)) & ((1 << 10) - 1);
+		/* Rebuild is being established on the server side, wait a bit longer */
+		if (err == -DER_UPDATE_AGAIN)
```

Comment:
not this patch issue, but it looks like DER_NOTLEADER should be treated same as UPDATE_AGAIN. i.e.

if (err == -DER_UPDATE_AGAIN || err == -DER_NOTLEADER)
             delay <<= 10;